### PR TITLE
add dztech/dz65rgbv3 keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
add dztech/dz65rgb keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz65rgbv3/dz65rgbv3.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+{
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void)
+{
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/dz65rgbv3.c:126]: the `develop` branch should be merged this weekend, and these will no longer be needed.
```suggestion
```

File: keyboards/dztech/dz65rgbv3/dz65rgbv3.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/dz65rgbv3.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dztech/dz65rgbv3/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dztech/dz65rgbv3/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/dztech/dz65rgbv3/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/rules.mk:11]:
```suggestion
```

File: keyboards/dztech/dz65rgbv3/readme.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# DZ65RGBv3
+
+A customizable 65% RGB keyboard.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: KBDFANS
+* Hardware Availability: [kbdfans](https://kbdfans.myshopify.com/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/dztech/dz65rgbv3/readme.md:9]: Could you add instructions on how to enter and use the bootloader to flash new firmware?

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
This should also be moved to `keyboards/dztech/dz65rgb/v3`.

**Pull request closed without merging**